tracpro/client.py

```diff
@@ -0,0 +1,53 @@
+from django.conf import settings
+from temba_client.clients import CursorQuery
+from temba_client.v2 import Boundary, TembaClient
+
+
+def get_client(org):
+    host = getattr(settings, 'SITE_API_HOST', None)
+    agent = getattr(settings, 'SITE_API_USER_AGENT', None)
+    return make_client(host, org.api_token, user_agent=agent)
+
+
+def make_client(host, api_token, user_agent):
+    # This method looks pointless, but it lets us mock it easily
+    # and change every call that returns a client.
+    return TracProClient(host, api_token, user_agent=user_agent)
+
+
+class TracProClient(TembaClient):
+    """
+    Customized TembaClient where the API calls that return multiple things
+    actually return them without having to tack on .all().
+
+    (Does this by returning a TracProCursorQuery instead of a CursorQuery.
+    Apart from being iterable-over, it should work the same.)
+    """
+    def _get_query(self, endpoint, params, clazz):
+        """
+        GETs a result query for the given endpoint
+        """
+        return TracProCursorQuery(self, '%s/%s.json' % (self.root_url, endpoint), params, clazz)
+
+    def get_boundaries(self):
+        return self._get_query('boundaries', {'geometry': 'true'}, Boundary)
+
+
+class TracProCursorQuery(CursorQuery):
+    """
+    Customized CursorQuery that allows iterating over it without having
+    to call `.all()` on it. More Pythonic.
```

Comment:
Right, this is definitely customized for our particular use case and not intended to be a fully general solution.

Thought I might note that you can also iterate over a Django queryset without having to call `.all()`.